sympy/physics/continuum_mechanics/column.py

```diff
@@ -0,0 +1,330 @@
+from sympy.solvers import linsolve, solve
+from sympy.core import Symbol, diff, symbols
+from sympy import dsolve, Function, Derivative, Eq, cos, sin, sqrt, tan
+from sympy.core.symbol import Dummy
+from sympy.printing import sstr
+
+class Column(object):
+    """
+    A column is a structural member designed to undertake axial
+    compressive loads. A column is characterized by its
+    cross-sectional profile(second moment of area), its length and
+    its material.
+
+    Examples
+    ========
+
+    There is a solid round bar 3 m long with second-moment I is used as a
+    column with both the ends pinned. Young's modulus of the Column is E.
+    The buckling load applied is 78KN
+
+    >>> from sympy.physics.continuum_mechanics.column import Column
+    >>> from sympy import Symbol, symbols
+    >>> E, I, P = symbols('E, I, P', positive=True)
+    >>> c = Column(3, E, I, 78000, top="pinned", bottom="pinned")
+    >>> c.end_conditions
+    {'bottom': 'pinned', 'top': 'pinned'}
+    >>> c.boundary_conditions
+    {'deflection': [(0, 0), (3, 0)], 'slope': [(0, 0)]}
+    >>> c.moment()
+    78000*y(x)
+    >>> c.solve_slope_deflection()
+    >>> c.deflection()
+    C1*sin(20*sqrt(195)*x/(sqrt(E)*sqrt(I)))
```

Comment:
Shouldn't the value of `C1` get evaluated using `boundary_conditions`?

Reply:
This is from the case where C1 and C2 comes out to be zero and hence even the deflection comes out to be zero. I have made a special case for it in the code.
Here C2 comes out to be zero and C1 remains unsolved.
I haven't found any resource of how to calculate the leftover value of C1, but I think maybe if the user provides some more boundary conditions we can solve it.
We could leave this for `solve_for_unknowns()` to do.